src/stan/lang/grammars/semantic_actions_def.cpp

```diff
@@ -2534,7 +2544,7 @@ namespace stan {
     boost::phoenix::function<set_int_range_upper> set_int_range_upper_f;
 
     void validate_int_data_expr::operator()(const expression& expr,
-                                            int var_origin, bool& pass,
+                                            var_origin var_origin, bool& pass,
```

Comment:
The variable `var_origin` should be passed by reference here.  It's also confusing to have the variable name and the type name be the same thing.  So how about `const var_origin& vo` instead?  Then all the `var_origin.` below get changed, which makes it less confusing as to value vs. data structure.